### PR TITLE
Adding CoC

### PR DESCRIPTION
**Pull request by kozross**
This is a first proposed draft of a Code of Conduct for Pext. I have included myself as a community organizer, as I am happy to be part of the enforcement team if requested. I've also licensed it consistently with #67, as I assume that change will go ahead eventually. If necessary, I can change the licensing to ``GPL3+`` to match, due to [one-way compatibility][1].

Sorry about the change in ``logo.svg`` - I don't even know why this registers as a change, since I literally didn't do anything to it content-wise.

[1]: https://creativecommons.org/2015/10/08/cc-by-sa-4-0-now-one-way-compatible-with-gplv3/

### CODE REVIEW

**Comment by TheLastProject** [comment]
I'll look at this later. For my mail, I guess just use "sylvia@hackerchick.me", it's my main mail.

On the note of logo.svg, you probably opened it in a text editor and hit save, and then added the file.

From a quick web search I found this, which seems a sensible way to undo that:
```
git reset HEAD^ pext/images/scalable/logo.svg
git commit --amend
git push --force
```

File: docs/CoC.md [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+
+We require all participants in our community to consider and abide by the
+principles above, in the spirit in which they are included in this Code of
+Conduct.
+
+## Expected behaviour
+
+The following behaviours are expected and requested of all community members:
+
+* Participate in an authentic and active way. In doing so, you contribute to the
```
Comment (TheLastProject) [inline docs/CoC.md]: Could you explain to me what "authentic and active" means? I'm not quite sure myself.

File: docs/CoC.md [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+  health and longevity of this community.
+* Exercise consideration and respect in your speech and actions.
+* Attempt collaboration before conflict.
+* Refrain from demeaning, discriminatory and harassing behaviour and speech.
+* Be mindful of your surroundings and your fellow participants. Alert community
+  leaders if you notice a dangerous situation, someone in distress, or
+  violations of this Safety and Well-Being Policy, even if they seem
```
Comment (TheLastProject) [inline docs/CoC.md]: I know you use Safety and Well-Being Policy in your version of this for your repo, but here it's called a Code of Conduct. Should probably use the term Code of Conduct here too.

File: docs/CoC.md [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+  orientation, ethnicity, religion or disability.
+* Inappropriate photography or recording. Note that photography or recording
+  without prior consent is _always_ inappropriate.
+* Inappropriate physical contact. You must have someone's consent before
+  touching them.
+* Unwelcome sexual attention. This includes, but is not limited to: sexualized
+  comments or jokes, inappropriate touching, groping and unwelcome sexual
+  advances. If you do not have someone's prior consent, then any kind of sexual
+  attention is unwelcome.
+* Deliberate intimidation, stalking or following (online or in person).
```
Comment (TheLastProject) [inline docs/CoC.md]: Hmm, I get what you mean here, but "following" sounds confusing as there is a follow button on GitHub. Maybe just simplify it to "Deliberate intimidation or stalking (online or in person)."?

File: docs/CoC.md [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+* Advocating for, or encouraging, or assisting in any of the above behaviour.
```
Comment (TheLastProject) [inline docs/CoC.md]: Seems like the first "or" is superfluous. Maybe even grammatically incorrect.

File: docs/CoC.md [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+unacceptable behaviour is expected to comply immediately.
+
+If a community member engages in unacceptable behaviour, the community
+organizers may take any action they deem appropriate. These include, but are not
+limited to:
+
+* A verbal warning (either in private or in public).
+* A temporary ban of any length with right of appeal.
+* A temporary ban of any length without right of appeal, by unanimous consent of
+  the community organizers.
```
Comment (TheLastProject) [inline docs/CoC.md]: This feels a bit... problematic to me, as Pext is pretty much only my project, and I don't want people to go and bother others because I won't listen to them. I'd rather have "with consent of the lead organizer" there. I know it is a bit of a "power grab" on my side, but well, it's my project in the end, and I want to be able to make the final decisions on abusive people.

File: docs/CoC.md [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+* A permanent ban with right of appeal.
+* A permanent ban without right of appeal, by unanimous consent of the community
+  organizers.
```
Comment (TheLastProject) [inline docs/CoC.md]: Same as comment above.

File: docs/CoC.md [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+* Temporary revocation of community organizer status for any length of time, with 
+  right of appeal.
+* Temporary revocation of community organizer status for any length of time,
+  without right of appeal, by unanimous consent of the remaining community
+  organizers.
```
Comment (TheLastProject) [inline docs/CoC.md]: Again, same as comment above.

File: docs/CoC.md [hunk 8]
```diff
@@ -0,0 +1,211 @@
...
+* Permanent revocation of community organizer status with right of appeal.
+* Permanent revocation of community organizer status without right of appeal, by
+  unanimous consent of the remaining community organizers.
```
Comment (TheLastProject) [inline docs/CoC.md]: Again same as above.

File: docs/CoC.md [hunk 9]
```diff
@@ -0,0 +1,211 @@
...
+* Removal of contributions to Pext from any Pext-associated repositories, with
+  right of appeal.
+* Removal of contributions to Pext from any Pext-associated repositories,
+  without right of appeal, by unanimous consent
```
Comment (TheLastProject) [inline docs/CoC.md]: Again, no unanimous consent please.

File: docs/CoC.md [hunk 10]
```diff
@@ -0,0 +1,211 @@
...
+* Which unacceptable behaviour(s) happened.
+* Any evidence for your report (such as screencaps, logs, etc)
+* Who (besides yourself) was affected by this behaviour.
+* Whether you want any additional actions taken by the community organizers to 
+  ensure that your safety and well-being remain intact.
+
+Additionally, community organizers are available to help community members
+engage with local law enforcement or otherwise help those experiencing
+unacceptable behaviour feel safe. In the context of in-person events, organizers
+will also provide escorts as desired by the person experiencing distress.
```
Comment (TheLastProject) [inline docs/CoC.md]: Uuuuhm. Like, I agree that this would be good! But I... don't know much about helping with local law enforcement and stuff. I think this paragraph should be removed because it's promising things that I don't know if I can keep. Sure, I'll always try to help, but I don't want to explicitly promising things I am not 100% sure I can keep.

File: docs/CoC.md [hunk 11]
```diff
@@ -0,0 +1,211 @@
...
+
+## Modifications
+
+The community organizers reserve the right to modify this Code of Conduct to 
+take into account future situations that could not have been foreseen
+at the time of its writing, or to address lack of clarity or inconsistencies
+that become apparent with practice. Additionally, we must update this document
+with any changes in the community organizer team when appropriate.
+
+We will notify the community when any changes are made, along with a description
```
Comment (TheLastProject) [inline docs/CoC.md]: Notify how?

**Comment by TheLastProject** [comment]
Also, please move the file to CODE_OF_CONDUCT.md in the root directory. GitHub only checks the root and .github directories for files like these, and otherwise it won't notify the user that we have one when making an issue.

File: docs/CoC.md (hunk 1 above)
Comment (kozross) [reply docs/CoC.md]: My take is 'don't be a troll' is the intended interpretation of 'authentic', and 'give back, don't just take' is the intended interpretation of 'active'.

File: docs/CoC.md (hunk 3 above)
Comment (kozross) [reply docs/CoC.md]: Yeah, that is a little bit vague and probably unnecessary. Will remove the 'following' part.

File: docs/CoC.md (hunk 5 above)
Comment (kozross) [reply docs/CoC.md]: 'Who watches the watchmen?' is a _very_ important question to ask with anything like this. You and I have both witnessed lead meltdowns which caused severe damage to the credibility and community of several projects. I don't want to assume that this will happen to you, but I prefer to have strong institutions in case something like this does happen. I will make this change if you request it, but I'm not sure I'd be good to serve on a community organization team in such a case.

File: docs/CoC.md (hunk 10 above)
Comment (kozross) [reply docs/CoC.md]: It basically means 'we will give any evidence we have to the LEOs if necessary, and we will advocate on your behalf'. You can't really do much more than that if you're not a lawyer, and I doubt anyone espects you to. Would a rephrase to something like that be OK?

File: docs/CoC.md (hunk 11 above)
Comment (kozross) [reply docs/CoC.md]: That's a very good question. Pext currently lacks a news 'blog', which would be the ideal place to have such notifications. Would this be something we could set up?

File: docs/CoC.md (hunk 5 above)
Comment (TheLastProject) [reply docs/CoC.md]: Well, I do understand that, but I do really want to stay in control of my own project, especially because practically 100% of the code, documentation and everything (except translations and themes) is written and maintained by me. I don't want to end up risking being alienated from my own project by ending in arguments on someone of the team unbanning someone I really want gone. If it really gets too bad, a hard fork as always possible. If enough of the community agrees they can always push me out of it. If you don't want to be part of it in this case, that's up to you.

File: docs/CoC.md (hunk 11 above)
Comment (VenomousDingo) [reply docs/CoC.md]: Most people just state "changes may be made to these regulations without notice" making it the user's responsibility to check.

File: docs/CoC.md (hunk 10 above)
Comment (VenomousDingo) [reply docs/CoC.md]: It's good to give evidence over to authorities but I don't think it's a good idea to become involved voluntarily in a legal case. Not all of us would be willing or able to.

File: docs/CoC.md (hunk 11 above)
Comment (kozross) [reply docs/CoC.md]: @VenomousDingo I think that's extremely user-unfriendly. It's probably one of my _biggest_ pet peeves with such documents.

File: docs/CoC.md (hunk 5 above)
Comment (VenomousDingo) [reply docs/CoC.md]: I have to agree on her argument to make final decisions as this is her project. She does reserve that right. If it was my project, I'd also be completely against sharing that power.

File: docs/CoC.md (hunk 10 above)
Comment (kozross) [reply docs/CoC.md]: Again, poor choice of words on my part. How about I rephrase this to:

"Community organizers are able to contact, and give evidence to, local law enforcement pertaining to any unacceptable behaviours you are subject to on your behalf. We are also able to help you otherwise feel safe."

File: docs/CoC.md (hunk 11 above)
Comment (VenomousDingo) [reply docs/CoC.md]: My main issue with it is that it adds unnecessary extra work and responsibilities. Usually, participation means automatic agreement.

Comment (kozross) [reply docs/CoC.md]: I think Pext having a news 'blog' is probably something that should exist in any case. I don't expect that such changes are common - mostly it'd be changes in community org people - which in my view doesn't place unreasonable overheads to running such a thing. I'm willing to remove this section if @TheLastProject agrees with you, however.

Comment (VenomousDingo) [reply docs/CoC.md]: I agree that there should be an update/announcement blog, yes. Like a Twitter account that can post commits or even be used as a "support chat" of sorts if someone can't access IRC.

Comment (kozross) [reply docs/CoC.md]: I'm happy to operate such an account, on whatever service(s), if this is what we decide to do. However, I think this is orthogonal and requires its own issue.

File: docs/CoC.md (hunk 10 above)
Comment (VenomousDingo) [reply docs/CoC.md]: What about: "Community organizers may assist local authorities and connect you with help or providing a safe environment."

File: docs/CoC.md (hunk 5 above)
Comment (kozross) [reply docs/CoC.md]: Just to clarify @TheLastProject, do you want decision control over _non-appealable_ enforcement actions, or _all_ enforcement actions? I'm not 100% clear on this.

File: docs/CoC.md (hunk 10 above)
Comment (kozross) [reply docs/CoC.md]: I'd be happy with that, except I'd use 'can' instead of 'may'.

Comment (VenomousDingo) [reply docs/CoC.md]: Sounds good to me.

File: docs/CoC.md (hunk 5 above)
Comment (TheLastProject) [reply docs/CoC.md]: Honestly, I want decisive control over every action. That doesn't mean I need to be consulted before every action. I just want some logs of who did what when for what reason (like, a Riot channel or GitHub team or whatever), so I know what's happening and can talk things through with a mod or reverse their action if I feel it's appropriate. Regardless of me reversing something, I feel we should have some central place where all moderation actions are logged anyway.

I know you're looking to create a democracy, but as long as still practically 100% of the entire project's code is mine, I am not ready to share Pext. The most share anyone else has in this project is @VenomousDingo by having created all themes and managing the Spanish translation, but even that is just a very small bit of Pext. At least for now, I want more of an "I'm the admin, others are mods" structure. Like OPs and half-ops on IRC, you know?

Comment (kozross) [reply docs/CoC.md]: Sure, and that's fine. I have made the necessary corrections, or at least I hope that the changes I've made reflect what you wanted to see.

File: CODE_OF_CONDUCT.md [hunk 12]
```diff
@@ -0,0 +1,200 @@
...
+
+* When and where the unacceptable behaviour occurred.
+* Who engages in unacceptable behaviour.
+* Which unacceptable behaviour(s) happened.
+* Any evidence for your report (such as screencaps, logs, etc)
+* Who (besides yourself) was affected by this behaviour.
+* Whether you want any additional actions taken by the community organizers to 
+  ensure that your safety and well-being remain intact.
+
+Additionally, community organizers can assist local authorities and connect you
```
Comment (TheLastProject) [inline CODE_OF_CONDUCT.md]: @VenomousDingo You said this word should be "can", right? I still feel "may" seems better because it doesn't seem to feel as much like a promise as "can"? Why do you feel can is better? Am I missing some grammar thing?

File: CODE_OF_CONDUCT.md [hunk 13]
```diff
@@ -0,0 +1,200 @@
...
+information:
+
+* Sylvia van Os <sylvia@hackerchick.me> (lead organizer)
+
+## Modifications
+
+The community organizers reserve the right to modify this Code of Conduct to 
+take into account future situations that could not have been foreseen
+at the time of its writing, or to address lack of clarity or inconsistencies
+that become apparent with practice. Additionally, we must update this document
```
Comment (TheLastProject) [inline CODE_OF_CONDUCT.md]: "must" sounds weird. Why not just "will"?

File: CODE_OF_CONDUCT.md [hunk 14]
```diff
@@ -0,0 +1,200 @@
...
+citizenship by encouraging participants to recognize and strengthen the
+relationships between our actions and their effects on our community.
+Communities mirror the societies in which they exist, and positive action is
+essential to counteract the many forms of inequality and abuses of power that
+exist in societies everywhere.
+
+As such, Pext is committed to the goal of open culture. Specifically, our
+community ensures that:
+
+* All source code is available under an open-source license.
```
[the remark this answers is not in the document]
Comment (kozross) [reply CODE_OF_CONDUCT.md]: Fair enough - I have something of a hyphen fetish, so it's probably just that. Will amend.

File: CODE_OF_CONDUCT.md [hunk 15]
```diff
@@ -0,0 +1,200 @@
...
+* Sustained disruption of community spaces or events.
+
+## Consequences of unacceptable behaviour
+
+Unacceptable behaviour from any community member, including anyone with
+decision-making authority, will not be tolerated. Anyone asked to stop
+unacceptable behaviour is expected to comply immediately.
+
+If a community member engages in unacceptable behaviour, the community
+organizers may take any action they deem appropriate, by agreement with, and
```
[the remark this answers is not in the document]
Comment (kozross) [reply CODE_OF_CONDUCT.md]: Yeah, that makes sense, will fix.

File: CODE_OF_CONDUCT.md [hunk 16]
```diff
@@ -0,0 +1,200 @@
...
+not limited to, guns, explosives (including fireworks without prior approval by
+a community organizer), and large knives such as those used for hunting or
+display, as well as any other item used for the purpose of causing injury and
+harm to others. Anyone seen in the possession of these items will be asked to
+leave immediately, and will only be allowed to return without the weapon.
+Community members are further required to comply with all local laws and
+regulations on this matter.
+
+Repeated violations of this policy are grounds for enforcement actions as if
+against unacceptable behaviour.
```
[the remark this answers is not in the document]
Comment (kozross) [reply CODE_OF_CONDUCT.md]: What I meant to say was 'repeatedly bringing guns into a community space is unacceptable behaviour', but to be honest, the weapons policy should just be put under unacceptable behaviour in the first place. The source material contained a separate section for this, so I'll just merge them.

File: CODE_OF_CONDUCT.md [hunk 17]
```diff
@@ -0,0 +1,200 @@
...
+concerns, please notify a community organizer as soon as possible. The current
+list of community organizers is:
+
+* Sylvia van Os <sylvia@hackerchick.me> (lead organizer)
+
+When reporting, please include the following information, to the best of your
+knowledge and ability:
+
+* When and where the unacceptable behaviour occurred.
+* Who engages in unacceptable behaviour.
```
[the remark this answers is not in the document]
Comment (kozross) [reply CODE_OF_CONDUCT.md]: Good catch on a bad typo.

File: CODE_OF_CONDUCT.md (hunk 12 above)
Comment (kozross) [reply CODE_OF_CONDUCT.md]: The use of 'may', to me at least, carries the interpretation 'if we feel like it', which I find to be totally against the spirit of this sentence. 'Can', on the other hand does not. I admit that it carries the interpretation of promising to do it, but not nearly as strongly IMHO. I welcome alternative phrasings, though.

File: CODE_OF_CONDUCT.md (hunk 13 above)
Comment (kozross) [reply CODE_OF_CONDUCT.md]: I'm trying to construct a justification for requiring an update schedule in the first place, and needing to fix the document in case of changes to the enforcement team is one such justification. Hence my use of 'must', in the sense of 'we have no choice but to'.

File: CODE_OF_CONDUCT.md (hunk 12 above)
Comment (TheLastProject) [reply CODE_OF_CONDUCT.md]: How about "may be able to"? It doesn't promise that we can help, but it does imply that we'll try to help if possible. Same for the "will also" a few sentences later, how about "will also try to"?

File: CODE_OF_CONDUCT.md (hunk 13 above)
Comment (TheLastProject) [reply CODE_OF_CONDUCT.md]: I think "must" just sounds weird. It implies we'd rather not. I'd just say "we will", it's a promise and it shows that we care to do so.

File: CODE_OF_CONDUCT.md (hunk 12 above)
Comment (kozross) [reply CODE_OF_CONDUCT.md]: To me, that sounds too watery. It's important that a document like this strikes a sufficiently imperative and competent tone to reassure people that we will act with proper due diligence when responding to unacceptable behaviour and making our community feel safe. Given that this is 2018, which means drive-by sexism and racism is unfortunately very common, and at least one woman in tech feels unsafe enough to _return to her own home_, I think we need to really stake out our position in much stronger language than what you are suggesting.

File: CODE_OF_CONDUCT.md (hunk 13 above)
Comment (kozross) [reply CODE_OF_CONDUCT.md]: I'll just redraft this whole section to cleanly separate 'why we need this' from 'what we will do about it', because I think this is the heart of this problem.

File: CODE_OF_CONDUCT.md [hunk 18]
```diff
@@ -0,0 +1,191 @@
...
+
+If a community member engages in unacceptable behaviour, the community
+organizers may take any action they deem appropriate, subject to correction 
+and reversal by the lead organizer. These include, but are not limited to:
+
+* A verbal warning (either in private or in public).
+* A temporary ban of any length, with or without right of appeal, from any or
+  all community spaces.
+* A permanent ban, with or without right of appeal, from any or all community
+  spaces.
```
Comment (TheLastProject) [inline CODE_OF_CONDUCT.md]: The Citizen Code of Conduct mentions no reimbursement when kicked out of paid events. I think this should be added, even though we don't have any yet, to be future-proof.

Comment (kozross) [reply CODE_OF_CONDUCT.md]: I agree, and will add.

File: CODE_OF_CONDUCT.md (hunk 12 above)
Comment (TheLastProject) [reply CODE_OF_CONDUCT.md]: The problem here is that "can" really promises we can help. I haven't ever worked with law enforcement. I don't want to promise I can get it right. Would "will attempt to assist local authorities to their best efforts" work? Because that I will obviously do, without doubt.

Comment (kozross) [reply CODE_OF_CONDUCT.md]: Sure, I can go with that. I'll make the necessary edits.

File: CODE_OF_CONDUCT.md [hunk 19]
```diff
@@ -0,0 +1,194 @@
...
+* Which unacceptable behaviour(s) happened.
+* Any evidence for your report (such as screencaps, logs, etc)
+* Who (besides yourself) was affected by this behaviour.
+* Whether you want any additional actions taken by the community organizers to 
+  ensure that your safety and well-being remain intact.
+
+Additionally, community organizers will assist local authorities and connect you
+with help or providing a safe environment inasfar as they are able. In the 
+context of in-person events, organizers will also provide escorts as desired by 
+the person experiencing distress.
```
Comment (TheLastProject) [inline CODE_OF_CONDUCT.md]: "inasfar" seems to not be a real word, I think you mean "insofar".

I also don't see the same "inasfar as they are able" for the escort part.

How about adding it to the last bullet point?

"Whether you want any additional actions taken by the community organizers to ensure that your safety and well-being remain intact. For example, if so desired and to the best of their ability, community organizers will assist local authorities, connect you with help, provide a safe environment or escort you in case of an in-person event."

Unless I missed why it's written as "additionally"?

File: CODE_OF_CONDUCT.md [hunk 20]
```diff
@@ -0,0 +1,194 @@
...
+* Sylvia van Os <sylvia@hackerchick.me> (lead organizer)
+
+When reporting, please include the following information, to the best of your
+knowledge and ability:
+
+* When and where the unacceptable behaviour occurred.
+* Who engaged in unacceptable behaviour.
+* Which unacceptable behaviour(s) happened.
+* Any evidence for your report (such as screencaps, logs, etc)
+* Who (besides yourself) was affected by this behaviour.
```
[the remark this answers is not in the document]
Comment (kozross) [reply CODE_OF_CONDUCT.md]: Agreed.

File: CODE_OF_CONDUCT.md [hunk 21]
```diff
@@ -0,0 +1,194 @@
...
+
+* Sylvia van Os <sylvia@hackerchick.me> (lead organizer)
+
+When reporting, please include the following information, to the best of your
+knowledge and ability:
+
+* When and where the unacceptable behaviour occurred.
+* Who engaged in unacceptable behaviour.
+* Which unacceptable behaviour(s) happened.
+* Any evidence for your report (such as screencaps, logs, etc)
```
[the remark this answers is not in the document]
Comment (kozross) [reply CODE_OF_CONDUCT.md]: Whoops. Will fix ASAP.

File: CODE_OF_CONDUCT.md (hunk 19 above)
Comment (kozross) [reply CODE_OF_CONDUCT.md]: Yes, I did; good catch, will amend. With regard to providing escorts at in-person events, I don't think this is something that should be up for a competency-based opt-out. With legal stuff, this is definitely somewhere we could (and should!) plead incompetence where appropriate, but setting up someone to accompany someone in distress isn't really something we can claim to be unable to do - if we fail there, we've failed basic organization.

The 'additionally' is from the original source. I agree that it's misplaced, and will remove it.

Comment (TheLastProject) [reply CODE_OF_CONDUCT.md]: While I get your point, I have a bit of an allergy for promising anything I'm not 100% sure I can do right. For example, if there will be a Pext event, it'll most likely only be me on the team. As soon as you have two people in distress, that can already become a problem, especially if those two don't get along well.

Personally, I find it worse to be promised with certainty something can be provided and then not get, then being told they will try their best so I can be mentally prepared for it possibly not going 100% perfectly.

Comment (kozross) [reply CODE_OF_CONDUCT.md]: If there will be a Pext event, it'll be:

1. So far into the future that who will be on the org team is very much up for debate; and
2. Under the auspices of a much larger conference, with its own Code of Conduct and its own org team.

Pext's community would have to grow by _at least_ an order of magnitude from where it currently is for an in-person event to even be a thing we can consider, with appropriate up-scaling of the org team.

Thus, I understand your concerns, but I still don't think this is something you need to worry about. My concern is that over-softening the text makes it come across as limp and ineffectual, which isn't something you want a Code of Conduct to be. I'll make the changes if you really insist, but I find them misplaced at best and actually detrimental to the document at worst, for no good reason.

Comment (TheLastProject) [reply CODE_OF_CONDUCT.md]: Well, here you are talking about the future. Why couldn't I just host a small Pext meetup at some Linux event with a few people? The Code of Conduct would already apply then. I don't think it's good to only look at the future and just say "Pext is too small for an event now, so it's not a problem the rules are broken for the current situation".

Comment (TheLastProject) [reply CODE_OF_CONDUCT.md]: @VenomousDingo What do you think about this whole thing?

Comment (kozross) [reply CODE_OF_CONDUCT.md]: Would said Linux event have its own Code of Conduct, with its own org team that you could ask for help? If so, you could always ask for assistance from them if this kind of issue comes up, as it's also their responsibility. I understand your reservations, but I honestly don't share them, precisely because the Pext community is currently far too small for its org team to be _unilaterally_ responsible for organizing and running an in-person event for it.

I really feel that softening this language would reduce confidence in the Code of Conduct for Pext. I understood the reservations (and made changes as appropriate) with matters of legal responsibility, because that's a serious issue on which pleading incompetence is not only easier for the organizers, but is actually the right thing to do. In this case, however, I really don't believe it's right to plead incompetence, precisely because I don't foresee any unilaterally-run Pext events until the community becomes so much larger than it currently is that it would necessitate considerable expansion of the org team. Until that point, you'd be part of a parent event anyway, whose team should (and would!) be available to assist with something like this, as that would be part of their jurisdiction also.

Comment (VenomousDingo) [reply CODE_OF_CONDUCT.md]: I just personally flat out disagree with promising anything. Using the words "may" or "can" should suffice but I don't like the usage of "will" in this at all. This can set us up for a legal mess in the future.

Comment (TheLastProject) [reply CODE_OF_CONDUCT.md]: @kozross, you said:
>precisely because I don't foresee any unilaterally-run Pext events until the community becomes so much larger than it currently is that it would necessitate considerable expansion of the org team

The @AsteroidOS community sometimes has small meetups "whereever the developer ends up being" (see https://twitter.com/AsteroidOS/status/897508413692923904). I really do want to organize small events like that, even while Pext is still fairly small. Which would already be in-person events.

Like, I would maybe consider making the language more certain when we have bigger events and a bigger team, but I want a Code of Conduct that will work now already too for the small events that may be hosted at some point.

**Comment by TheLastProject** [comment]
It looks pretty much fine to me now. Just a few last questions:

- You use `<email@domain.tld>`, but GitHub makes the `<` and `>` disappear. Maybe just use ( and ) on the e-mail place instead?
- You no longer want to be on the organizer team yourself, now that I pretty much am BDFL, right? Just double-checking.
- @VenomousDingo, do you have any last comments? How do you feel about joining the team here?

**Comment by kozross** [comment]
You are correct, yes - you are Pext's BDFL as per the current draft.

File: CODE_OF_CONDUCT.md [hunk 22]
```diff
@@ -0,0 +1,194 @@
...
+* Inappropriate photography or recording. Note that photography or recording
+  without prior consent is _always_ inappropriate.
+* Inappropriate physical contact. You must have someone's consent before
+  touching them.
+* Unwelcome sexual attention. This includes, but is not limited to: sexualized
+  comments or jokes, inappropriate touching, groping and unwelcome sexual
+  advances. If you do not have someone's prior consent, then any kind of sexual
+  attention is unwelcome.
+* Deliberate intimidation or stalking (online or in person).
+* Sustained disruption of community spaces or events. This includes brining
```
Comment (VenomousDingo) [inline CODE_OF_CONDUCT.md]: Should be "bringing" not "brining" here.

File: CODE_OF_CONDUCT.md [hunk 23]
```diff
@@ -0,0 +1,194 @@
...
+* Permanent revocation of community organizer status with or without right of 
+  appeal.
+* Removal of contributions to Pext from any Pext-associated repositories, with
+  or without right of appeal.
+* Notification of law enforcement, or other appropriate authorities, along with
+  any evidence of wrongdoing provided to the community organizers.
+
+## Reporting guidelines
+
+If you are subject to, or witness, unacceptable behaviour, or have any other
```
Comment (VenomousDingo) [inline CODE_OF_CONDUCT.md]: Should be "or witness of,".

File: CODE_OF_CONDUCT.md [hunk 24]
```diff
@@ -0,0 +1,194 @@
...
+concerns, please notify a community organizer as soon as possible. The current
+list of community organizers is:
```
Comment (VenomousDingo) [inline CODE_OF_CONDUCT.md]: Should be "are:" not "is:", even if there's only one person on the list so far.

File: CODE_OF_CONDUCT.md [hunk 25]
```diff
@@ -0,0 +1,194 @@
...
+* When and where the unacceptable behaviour occurred.
+* Who engaged in unacceptable behaviour.
+* Which unacceptable behaviour(s) happened.
+* Any evidence for your report (such as screencaps, logs, etc).
+* Who (besides yourself) was affected by this behaviour, if applicable.
+* Whether you want any additional actions taken by the community organizers to 
+  ensure that your safety and well-being remain intact.
+
+Community organizers can assist local authorities and connect you
+with help or providing a safe environment insofar as they are able. In the
```
Comment (VenomousDingo) [inline CODE_OF_CONDUCT.md]: "Insofar as" should be "provided".

File: CODE_OF_CONDUCT.md [hunk 26]
```diff
@@ -0,0 +1,194 @@
...
+* Any evidence you can provide for your grievance.
+* What actions you wish to be taken to address your grievance if found genuine.
+
+If you have been temporarily, or permanently, banned without right of appeal,
+any grievances filed with community organizers regarding the incident which
+caused the ban will be ignored.
+
+## Scope
+
+We expect all community participants (contributors, paid or otherwise; sponsors;
```
Comment (VenomousDingo) [inline CODE_OF_CONDUCT.md]: "paid or otherwise" does not need to be stated, and the rest of the semicolons can be changed to commas.

File: CODE_OF_CONDUCT.md [hunk 27]
```diff
@@ -0,0 +1,194 @@
...
+and other guests) to abide by this Code of Conduct in all 
+community venues, as well as in one-to-one communications pertaining to community
```
Comment (VenomousDingo) [inline CODE_OF_CONDUCT.md]: The comma does not to be here.

File: CODE_OF_CONDUCT.md [hunk 28]
```diff
@@ -0,0 +1,194 @@
...
+If you have been temporarily, or permanently, banned without right of appeal,
+any grievances filed with community organizers regarding the incident which
+caused the ban will be ignored.
+
+## Scope
+
+We expect all community participants (contributors, paid or otherwise; sponsors;
+and other guests) to abide by this Code of Conduct in all 
+community venues, as well as in one-to-one communications pertaining to community 
+business. This Code of Conduct also applies to unacceptable behaviour occuring
```
Comment (VenomousDingo) [inline CODE_OF_CONDUCT.md]: "This Code of Conduct, within reason, also"

File: CODE_OF_CONDUCT.md [hunk 29]
```diff
@@ -0,0 +1,194 @@
...
+
+## Modifications
+
+This Code of Conduct is a living document, designed to serve the needs of our
+community at the present time, but also in the future. To ensure that it remains
+relevant, to address future situations that cannot be foreseen at the time of
+its writing, and to update the community organizer list, this document will
+occasionally receive updates.
+
+We will notify the community when any changes are made, along with a description
```
Comment (VenomousDingo) [inline CODE_OF_CONDUCT.md]: Should be "We will do our best to notify".

File: CODE_OF_CONDUCT.md (hunk 23 above)
Comment (richarson) [reply CODE_OF_CONDUCT.md]: I just want to point out that witness is also a verb, so "if you witness unacceptable behaviour" could be valid.
Sorry for hijacking :)

File: CODE_OF_CONDUCT.md (hunk 24 above)
Comment (richarson) [reply CODE_OF_CONDUCT.md]: I believe it refers to "list", not "organizers", so "is" might actually be correct.
Or maybe "The current community organizers are:" ?

**Comment by TheLastProject** [comment]
Well, this seems fine then.

@kozross Could you add "Emily Lau (fuchslein@hackerchick.me)" to the list of community organizers below me?

@richarson I saw your comments, they seem to be resolved already :)

File: CODE_OF_CONDUCT.md [hunk 30]
```diff
@@ -0,0 +1,197 @@
...
+* When and where the unacceptable behaviour occurred.
+* Who engaged in unacceptable behaviour.
+* Which unacceptable behaviour(s) happened.
+* Any evidence for your report (such as screencaps, logs, etc).
+* Who (besides yourself) was affected by this behaviour, if applicable.
+* Whether you want any additional actions taken by the community organizers to 
+  ensure that your safety and well-being remain intact.
+
+Community organizers can assist local authorities and connect you
+with help or providing a safe environment provided they are able. In the
```
Comment (TheLastProject) [inline CODE_OF_CONDUCT.md]: I cannot parse this sentence anymore, afraid this one is going to need a rephrase for readability purposes. I'm pretty sure this is the last ever thing I'll have to ask.

Comment (TheLastProject) [reply CODE_OF_CONDUCT.md]: How about "Provided they are able, community organizers can assist local authorities, connect you with help or provide a safe environment."? I think that's what you meant to say here?

**Comment by TheLastProject** [comment]
Well, we finally got it merged, thanks for everything! :)
